### PR TITLE
map macOS compat 10.16 version to 11.0

### LINKED ISSUE #41012: Environment.OSVersion returns wrong info on Big Sur
This returns 10.16 and _may_ be caveat of the beta. 
However, is is curious that OS tools report 11.0 as expected:
```
furt@macos-11 foo % sysctl kern.osproductversion
kern.osproductversion: 11.0

furt@macos-11 foo % sw_vers
ProductName:	macOS
ProductVersion:	11.0
BuildVersion:	20A5343i
```
This seems to also impact `dotnet --info` and perhaps package selection.

cc: @akoeplinger @marek-safar @danmosemsft

### PR DESCRIPTION
**Pull request by wfurt**
It seems like we get "wrong" os version because we build and link against older macOS SDK. 
With that we get 10.16 instead of 11.0 as expected. 

For time being this adds mapping from 10.16 to 11.0. 
It also updates RID calculation. That one is based on kernel version.
11.0 ships with 20.0 and the current logic needed to be updated as well. 

fixes  #41012

### CODE REVIEW

**Review by danmoseley — APPROVED** [review]

File: src/installer/corehost/cli/hostmisc/pal.unix.cpp [hunk 1]
```diff
@@ -562,18 +561,31 @@ pal::string_t pal::get_current_os_rid_platform()
         size_t pos = release.find('.');
         if (pos != std::string::npos)
         {
-            // Extract the major version and subtract 4 from it
-            // to get the Minor version used in OSX versioning scheme.
-            // That is, given a version 10.X.Y, we will get X below.
-            int minorVersion = stoi(release.substr(0, pos)) - 4;
-            if (minorVersion < 10)
+            int majorVersion = stoi(release.substr(0, pos));
```
Comment (scalablecory) [inline src/installer/corehost/cli/hostmisc/pal.unix.cpp:564]: Can the integer be negative? Should this be `stoul`?

**Review by scalablecory — APPROVED** [review]
lgtm though I'm not familiar with the MacOS APIs.

**Comment by directhex** [comment]
This seems to have caused a new regression for me?

https://helix.dot.net/api/2019-06-17/jobs/0c4ba9a4-e6f1-4153-a2dd-257ca6e923cf/workitems/System.Runtime.Extensions.Tests/console

**Comment by wfurt** [comment]
Is that executed on macOS 11? I cannot tell from the log @directhex

**Comment by directhex** [comment]
`"QueueId": "osx.1100.amd64.open",`

**Comment by jkotas** [comment]
Some tests setups use stale `dotnet` that does not have this change. This will fix itself once the change propagates through the system. We may want to relax the test in the  meantime.

**Comment by wfurt** [comment]
I'll check if the test pass locally. I did not run this test set before.

**Comment by directhex** [comment]
@wfurt how did it look locally? I'm still getting errors out of Helix

**Comment by wfurt** [comment]
It fails for me locally as well @directhex. But if I replace `libnethost.dylib` for test run with the one built by installer, then test can pass. (and I get osx.11.0 rid). So I think @jkotas is right and this will eventually resolve it self. 
I din't know if we get this file from published package or dotnet used to build the test.

**Comment by danmosemsft** [comment]
I think we probably want to propose this change to Tactics for 5.0, since it's very localized, and to accommodate the OS which will be released 1 year before 6.0.

**Comment by danmosemsft** [comment]
Never mind- it's already there.

**Pull request merged**